prepare-image.sh

```diff
@@ -2,75 +2,58 @@
 
 set -euxo pipefail
 
-IRONIC_PKG_LIST="/tmp/ironic-${INSTALL_TYPE}-list"
-
 echo "install_weak_deps=False" >> /etc/dnf/dnf.conf
 # Tell RPM to skip installing documentation
 echo "tsflags=nodocs" >> /etc/dnf/dnf.conf
 
 dnf install -y 'dnf-command(config-manager)'
 
-# NOTE(elfosardo): building the container using ironic RPMs is
-# now deprecated and it will be removed in the future.
-# RPM install #
-if [[ "$INSTALL_TYPE" == "rpm" ]]; then
-    curl -o /etc/yum.repos.d/delorean.repo https://trunk.rdoproject.org/centos9-master/puppet-passed-ci/delorean.repo && \
-    curl -o /etc/yum.repos.d/delorean-deps.repo https://trunk.rdoproject.org/centos9-master/delorean-deps.repo
-
-    # NOTE(elfosardo): enable CRB repo for more python3 dependencies
-    dnf config-manager --set-enabled crb
-    dnf upgrade -y
-    xargs -rtd'\n' dnf install -y < "$IRONIC_PKG_LIST"
+# emulate uid/gid configuration to match rpm install
+IRONIC_UID=997
+IRONIC_GID=994
+BUILD_DEPS="python3-devel gcc git-core python3-setuptools python3-jinja2"
```

Comment:
If we are touching these lines now, let's use modern syntax and avoid shellcheck ignores later on. Also makes cleaner PRs going forward. Also sorted.

```suggestion
declare -a BUILD_DEPS=(
    gcc 
    git-core 
    python3-devel 
    python3-jinja2
    python3-setuptools 
)
```